src/Features/LanguageServer/Protocol/Handler/RequestExecutionQueue.cs

```diff
@@ -48,18 +96,18 @@ namespace Microsoft.CodeAnalysis.LanguageServer.Handler
     /// more messages, and a new queue will need to be created.
     /// </para>
     /// </remarks>
-    internal partial class RequestExecutionQueue
+    public abstract partial class RequestExecutionQueue<RequestContextType> : IRequestExecutionQueue<RequestContextType> where RequestContextType : struct
```

Comment:
Why not have the execution queue in the CLaSP framework? Is it due to how you create the queue items? I feel like this is where the magic exists / what we'd want the CLaSP bit to handle

Reply:
Yeah I agree we'll want to make a "base" `RequestExecutionQueue` available, but I I need to noodle a bit on the proper abstractions and I wanted to get this out in front of people.

Reply:
coo coo. Ya it's probably one of the most interesting "design" pieces if you ask me. Aka how you define an extensible system that can be used to update both Roslyn's workspace system and Razor's project snapshot manager systems (aka some random document state manager).